File: jscomp/test/gpr_4519_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = Mt.eq_suites ~test_id ~suites loc x y 


type t =
  | Required
  | Optional
  
let nextFor (x : t option) =
  match x with
  | Some Required -> Some Optional
  | Some Optional -> None
  | None  -> Some Required


;; eq __LOC__   (nextFor (Some Required)) (Some Optional)
let () = Mt.from_pair_suites __MODULE__ !suites

File: jscomp/test/gpr_4519_test.js
'use strict';

var Mt = require("./mt.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  return Mt.eq_suites(test_id, suites, loc, x, y);
}

function nextFor(x) {
  if (x !== undefined) {
    if (x) {
      return ;
    } else {
      return /* Optional */1;
    }
  } else {
    return /* Required */0;
  }
}

eq("File \"gpr_4519_test.ml\", line 17, characters 6-13", /* Optional */1, /* Optional */1);

Mt.from_pair_suites("Gpr_4519_test", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.nextFor = nextFor;
/*  Not a pure module */
